### ISSUE TITLE
Sitemap.xml builder find hidden root pages

### ISSUE THREAD

**Issue by ecentral** [mblaschke/TYPO3-metaseo#272]
We are using a multi-domain typo3 6.2.29 system with more then 50 domains. The task "Sitemap.xml builder" also finds hidden rootpages. We used version 2.0.1.

Reason:
The class "getRootPages" in Metaseo\Metaseo\Scheduler\Task\AbstractTask dont ignore hidden pages.

Fix:
Edit this function and add a "AND hidden=0" tho sql query.

```
/**
 * Scheduler Task Sitemap Base
 */
abstract class AbstractTask extends T3AbstractTask
{
...
    /**
     * Get list of root pages in current typo3
     *
     * @return  array
     */
    protected function getRootPages()
    {
        $query = 'SELECT uid
                    FROM pages
                   WHERE is_siteroot = 1
                     AND deleted = 0';

        return DatabaseUtility::getColWithIndex($query);
    }
```

**Comment by cedricziel** [comment]
Fixed in #273

**Comment by ecentral** [comment]
Thx 👍

**Comment by cedricziel** [comment]
No problem. Easy pick. Let's hope for a merge.

**Comment by thomaszbz** [comment]
For the merge, I'll reopen this issue. Will be closed when merge is done.

**Comment by thomaszbz** [comment]
Scheduler/Task/AbstractTask is only implemented/used by SitemapTxtTask and SitemapXmlTask.

Basically, that means that we use it for _generation_ of sitemaps only. It is never used for editing/showing something in the backend etc.

However, we should still consider hidden root pages for generation of sitemaps because even if a root page is hidden, it could easily happen that subpages of the root page are _supposed_ to be indexed and rendered in a sitemap.

In the sitemap, of course, hidden pages should not occur (no matter if root page or normal page).

@ecentral if you want to disable a root page, I recommend to do that in Metaseo's control center:
![auswahl_065](https://cloud.githubusercontent.com/assets/11200727/20870308/13c0c3f8-ba86-11e6-9a09-735ab966b0a8.png)

**Comment by cedricziel** [comment]
I would like to chime in on this. It's your decisionto keep a counter-intuitive behaviour. - Yet you should think about if what you want should rather be a whitelist, not a blacklist. Disabling pages (and records in general) is **the** way TYPO3 normally disables access to rows and I can't wrap my head around why you would choose the opposite of how the system works in every other part.

**Comment by thomaszbz** [comment]
@cedricziel I assume that a lot of users hide pages but still want http-access to their subpages. Isn't that very common?

Now, it's hard to explain users why a hidden root page should have a different effect on the sitemap than any other hidden page. It simply would not be consistent.

It's about the whole tree under the root page, not about the root page itself.

On the other side, it's just too easy: 2-3 clicks and it's done. Has always been. Should always be.

**Comment by cedricziel** [comment]
Yep, you're right. it's about consistency. The behaviour you describe is mostly about "hidden in menu".

Deactivating a root-page does exactly what is says: It disables the site underneath until it hits the next root page.

It's about being precise and correct for those users that care or have bigger installations than those single-tree installations. @ecentral is obviously one of those power users and expects to have a certain level of consistency.

Yep, it's 2 or 3 clicks right *after* you discover that the extension behaves different from every other extension.

Thank you nevertheless for taking on the discussion. I don't think we'll get to a consensus ;)

**Comment by thomaszbz** [comment]
@cedricziel Even if, I could not just change behaviour when users rely on a behaviour as it is. I mean that would be somehow breaking. Breaking a lot. But this is not the issue if there was a good reason for it. There's not a single line of holy code in Metaseo (just a lot of old code).

All l can say is that the existing behaviour is intentional, therefore it's not a bug, therefore not bugfixed.

For considering it as a feature request (which breaks existing behaviour), I can see at least disadvantages when people rely on existing behaviour. That should be and stay the default behaviour.

In respect to "those single tree installations" I don't go along with your consideration. Especially not in respect to Metaseo.

If you want to have it "fixed" and it is important to you, maybe we can discuss about a config variable which is well-documented, but disabled by default. Just to offer a peace deal.

**Comment by thomaszbz** [comment]
In respect to implementation, instead of using a config variable, maybe it should be exposed to the GUI. E.g. something like

- [ ] Sitemap active
- [ ] Automatically bind to page status

and the setting should make it to metaseo's database tables. Eventually, there's arbitrary places we use it.

To trigger the setting, eventually hooking into TYPO3 would be the way to go.
